### PR TITLE
{ts} test test test

### LINKED ISSUE #19: Tests: to do
Ask about the following:
- [ ] What is the input to `get_parsimony_pep2prot()`?
- [x] How were the `.rds` files in `/data` generated?
- [ ] silac_psm_data is `data.frame` not `MSnSet`?

Do the following:
- [ ] Add some peptide level SILAC data in `/extdata` for testing `estimate_incorporation()`. Also need this for getting 100% test coverage on `parse_features()`.
- [x] Have a look at the `.rds` files in `/tests/testthat/`. Need to move them into their own folder to reduce clutter.
- [x] Make the names and internal structure of test files consistent.

Need help with the following:
- [x] Add a phospho PD output in `/inst/extdata` to test some of the functions in `ptm.R`.
- [ ] Understand what the functions in `/R/combine_runs.R` do and write tests for them.
- [x] Writing a test for `restrict_features_per_protein()`.

### PR DESCRIPTION
**Pull request by csdaw**
- Adds lots of tests!
- Involves quite a bit of file renaming and moving around
- Replaces Travis with GitHub Actions

Now everything has at least 1 test, except `estimate_incorporation()` because I couldn't get it to work.

Closes #19 and #16

### CODE REVIEW

**Comment by csdaw** [comment]
I didn't mean to merge master. I'm going to do a bad thing and remove the last commit (i.e. rewriting history).

File: R/qc_plots.R [hunk 1]
```diff
@@ -12,6 +12,9 @@
 plot_quant <- function(obj,
                        method=c('box', 'density', 'histogram'),
                        facet_by_sample=FALSE){
+  # check method argument
+  method = match.arg(method)
```
Comment (TomSmithCGAT) [inline R/qc_plots.R:16]: Ah nice. I should have know about this function already!

File: tests/testthat/test-get_ratio.R [hunk 2]
```diff
@@ -27,3 +27,5 @@ test_that("there are no NAs in the new 'missing' column", {
 
   expect_false(any(is.na(df2$missing)))
 })
+
+#### Sanity checks -------------------------------------------------------------
```
Comment (TomSmithCGAT) [inline tests/testthat/test-get_ratio.R:31]: Can be deleted?

File: tests/testthat/test-go.R [hunk 3]
```diff
@@ -1,21 +1,181 @@
...
+
+#### Sanity checks -------------------------------------------------------------
+
+test_that("determine_ancestor_function produces NA is ontology is NA", {
+  result <- determine_ancestor_function(rbp_terms[1], NA)
+  expect_identical(result, NA)
+})
+
+test_that("determine_ancestor_function produces NA is ontology is nonsense", {
+  result <- determine_ancestor_function(rbp_terms[1], "banana")
```
Comment (TomSmithCGAT) [inline tests/testthat/test-go.R:126]: Someone's been having too much fun testing :banana:

File: tests/testthat/test-tmt_qc_plots.R [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+# Load some TMT PSM level data for tests
+load(system.file("testdata", "small_psm_tmt_phospho.rda",
+                 package = "camprotR"))
+load(system.file("testdata", "small_psm_tmt_total.rda",
+                 package = "camprotR"))
+
+#### Tests ---------------------------------------------------------------------
+# bit of a hacky test because can't use expect_equal_to_reference as the
+# object changes slightly every time you run the test.
```
Comment (TomSmithCGAT) [inline tests/testthat/test-tmt_qc_plots.R:13]: There's a package for regression testing for images but seems like overkill to me (https://github.com/r-lib/vdiffr). I'm happy with the 'hacky' test

File: tests/testthat/test-utilities.R [hunk 5]
```diff
@@ -12,3 +12,5 @@ test_that("remove_dots works", {
   colnames(df) <- remove_dots(colnames(df))
   expect_equal(grep("\\.", colnames(df)), 1)
 })
+
+#### Sanity checks -------------------------------------------------------------
```
Comment (TomSmithCGAT) [inline tests/testthat/test-utilities.R:16]: Again, delete?

**Review by TomSmithCGAT — APPROVED** [review]
Nothing major to add. All looks good to me

**Comment by TomSmithCGAT** [comment]
I suggest merging and opening a new issue re `estimate_incorporation()`, which can be assigned to me.

**Comment by TomSmithCGAT** [comment]
Nice work BTW :muscle:

File: .github/workflows/check-bioc.yml [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+
+## Acronyms:
+## * GHA: GitHub Action
+## * OS: operating system
+
+on:
+  push:
+    branches:
+      - master
+      - '{ts}_test_test_test'
```
[the remark this answers is not in the document]
Comment (csdaw) [reply .github/workflows/check-bioc.yml:27]: Yes!

File: tests/testthat/test-get_ratio.R (hunk 2 above)
Comment (csdaw) [reply tests/testthat/test-get_ratio.R:31]: Yes again

File: tests/testthat/test-go.R (hunk 3 above)
Comment (csdaw) [reply tests/testthat/test-go.R:126]: I think you'll find some more fruit names scattered here and there 😆

File: tests/testthat/test-utilities.R (hunk 5 above)
Comment (csdaw) [reply tests/testthat/test-utilities.R:16]: Yes!

**Pull request merged**